substra/sdk/backends/remote/backend.py

```diff
@@ -109,26 +113,103 @@ def add(self, spec, exist_ok=False, spec_options=None):
             response = self._add(asset_type, data, files=files, exist_ok=exist_ok)
         # The backend has inconsistent API responses when getting or adding an asset
         # (with much less data when responding to adds).
         # A second GET request hides the discrepancies.
+        # Do not do this with a compute plan or we lose the id_to_key field
         if asset_type == schemas.Type.ComputePlan:
             return response
 
         key = _get_asset_key(response)
         return self.get(asset_type, key)
 
-    def update_compute_plan(self, compute_plan_id, spec):
-        return self._client.request(
-            'post',
-            schemas.Type.ComputePlan.to_server(),
-            path=f"{compute_plan_id}/update_ledger/",
-            json=spec.dict(exclude_none=True),
+    def _auto_batching_compute_plan(self,
+                                    spec,
+                                    batch_size,
+                                    compute_plan_id=None,
+                                    exist_ok=None,
+                                    spec_options=None):
+        """Auto batching of the compute plan tuples
+        """
+        spec_options = spec_options or dict()
+        batches = compute_plan.auto_batching(
+            spec,
+            is_creation=compute_plan_id is None,
+            batch_size=batch_size,
         )
 
+        # Create / update by batch
+        id_to_keys = dict()
+        asset = None
+        for tmp_spec in batches:
+            if compute_plan_id:
+                asset = self._client.request(
+                    'post',
+                    schemas.Type.ComputePlan.to_server(),
+                    path=f"{compute_plan_id}/update_ledger/",
+                    json=tmp_spec.dict(exclude_none=True),
+                )
+                id_to_keys.update(asset['IDToKey'])
+            else:
+                with tmp_spec.build_request_kwargs(**spec_options) as (data, files):
+                    asset = self._add(
+                        schemas.Type.ComputePlan,
+                        data,
+                        files=files,
+                        exist_ok=exist_ok
+                    )
+                compute_plan_id = asset['computePlanID']
+            id_to_keys.update(asset['IDToKey'])
+
+        # Special case: no tuples
+        if asset is None:
+            spec_options[AUTO_BATCHING] = False
+            spec_options[BATCH_SIZE] = batch_size
+            if compute_plan_id:
+                return self.update_compute_plan(
+                    compute_plan_id=compute_plan_id,
+                    spec=spec,
+                    spec_options=spec_options
+                )
+            else:
+                return self.add(spec=spec, exist_ok=exist_ok, spec_options=spec_options)
+
+        asset['IDToKey'] = id_to_keys
+        return asset
+
+    def update_compute_plan(self, compute_plan_id, spec, spec_options=None):
```

Comment:
I don't think we should support the autobatching for the update_compute_plan method.

This method has been created to support mannually the autobatching and to be able to create big compute plans. Updating existing compute plans is not supported by the platform as we cannot update a done compute plan.

In my opinion, we should warn the user in the method documentation (or say it's deprecated) and tell him to prefer the `create_compute_plan()` method.

Reply:
For the case where users would prefer adding everything at once, it is redundant indeed.  
Weren't there cases where the user did not know when to stop the compute plan (one tuple = one epoch of training for example) and they would update it continuously ?

Reply:
Yes, you're right. It makes sense.

This use case is not fully supported by the platform (local folders deleted when compute plan is done by default) but this is an other story and should be treated separately.